api/src/main/java/com/datastrato/gravitino/rel/expressions/Indexes/Index.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.rel.expressions.Indexes;
+
+/** Represents an index on a table. */
+public interface Index {
+
+  IndexType type();
+
+  /** @return The name of the index. */
+  String name();
+
+  /** @return Returns the field name under the table contained in the index. eg: table.id */
+  String[][] fieldNames();
+
+  enum IndexType {
+    PRIMARY_KEY,
+    UNIQUE_KEY,
```

Comment:
If the index is neither `PRIMARY_KEY` or `UNIQUE_KEY`, how do we express it?

Reply:
Can you please refer to Spark's `TableIndex` design? I'm wondering do we need to add more interfaces like properties to make it more flexible.

Reply:
Seems there are some different scenarios between Spark and Gravitino, Limited by spark SQL dialect，it couldn't specify index when creating tables, so it provides index operate interface like createIndex, dropIndex, etc, as I understand